src/main/java/ai/levo/LevoSatelliteService.java

```diff
@@ -1,11 +1,55 @@
 package ai.levo;
 
-import okhttp3.ResponseBody;
-import retrofit2.Call;
-import retrofit2.http.Body;
-import retrofit2.http.POST;
-
-public interface LevoSatelliteService {
-    @POST("/1.0/ebpf/traces")
-    Call<ResponseBody> sendHttpMessage(@Body HttpMessage message);
+import ai.levo.exceptions.SatelliteMessageFailed;
+import burp.IBurpExtenderCallbacks;
+import burp.IExtensionHelpers;
+import burp.IHttpRequestResponse;
+import burp.IHttpService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LevoSatelliteService {
+
+    public static LevoSatelliteService create(String satelliteUrl, IBurpExtenderCallbacks callbacks) throws MalformedURLException {
+        return new LevoSatelliteService(callbacks, satelliteUrl);
+    }
+
+    private final IExtensionHelpers helpers;
+    private final IBurpExtenderCallbacks callbacks;
+
+    private final IHttpService service;
+
+    public LevoSatelliteService(IBurpExtenderCallbacks callbacks, String satelliteUrl) throws MalformedURLException {
+        this.helpers = callbacks.getHelpers();
+        this.callbacks = callbacks;
+        var url = new URL(satelliteUrl);
+        this.service = helpers.buildHttpService(url.getHost(), url.getPort(), url.getProtocol().equals("https"));
+    }
+
+    public IHttpRequestResponse sendHttpMessage(HttpMessage httpMessage) throws SatelliteMessageFailed, JsonProcessingException {
+        var mapper = new ObjectMapper();
+        var jsonBody = mapper.writeValueAsString(httpMessage);
+        byte[] body = helpers.stringToBytes(jsonBody);
+        List<String> newHeaders = new ArrayList<>();
+        newHeaders.add("POST /1.0/ebpf/traces HTTP/1.1");
```

Comment:
Why is this needed @mgiamberardino ?

Reply:
@buchi-busireddy This is the way to define the path of the endpoint we are trying to hit.